lucene/core/src/java/org/apache/lucene/index/SegmentInfo.java

```diff
@@ -68,7 +69,7 @@
 
   private Map<String,String> diagnostics;
   
-  private final Map<String,String> attributes;
+  private Map<String,String> attributes;
```

Comment:
Do we need to make it volatile so that toString sees the up-to-date reference to the attributes?

Reply:
I don't think this is helping anybody. This entire class is mutable and accessed without locking. I think we are making the situation as good as it gets without lock since we always safe-publish with a new instance. There are many other things like `maxDoc` and `isCompoundFile` that have the same issues.